### PR TITLE
Build Docker image

### PR DESCRIPTION
**Pull request by trcywu**

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM node:12
+
+# Setting working directory. All the path will be relative to WORKDIR
+WORKDIR /usr/src/app
+
+# Installing dependencies
+COPY package*.json ./
+RUN npm install
```
Comment (frigus02) [inline frontend/Dockerfile:8]: We're using `yarn` and we have `yarn.lock` file. Let's use that here, too.
```suggestion
COPY package.json yarn.lock ./
RUN yarn install --frozen-lockfile
```

Same further down in the file.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -31,3 +30,23 @@ jobs:
           yarn lint
           yarn test
           yarn build
```
Comment (frigus02) [inline .github/workflows/nodejs.yml:32]: Let's move these steps into the Docker build, too. This way we only run them once.

This means we don't even need to setup node on the build agent, because everything runs in Docker.

File: frontend/manifests/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          image: citracytest.azurecr.io/frontend
+          ports:
+            - containerPort: 3000
+          resources:
+            limits:
+              cpu: 100m
+              memory: 100Mi
+            requests:
+              cpu: 10m
+              memory: 100Mi
```
Comment (frigus02) [inline frontend/manifests/deployment.yml:26]: We might want to remove those, but we can do it later. I imagine the node app could require more resources than this.

File: infrastructure/README.md [hunk 4]
```diff
@@ -6,127 +6,129 @@ We use [Terraform](https://www.terraform.io/) to build our environments.
...
 1. To reduce infrastructure costs for the NHS, please destroy your environment when you no longer need it.
 
-    ```bash
-    terraform destroy
-    ``` 
+   ```bash
+   terraform destroy
+   ```
+
+   At the same time, delete ./infrastructure/environments/dev/.terraform folder as it caches your existing terraform setup and if left untouched will create an error next time you try to spin up a new cluster.
```
Comment (frigus02) [inline infrastructure/README.md:134]: That does not happen on my machine. This should not be necessary. Deleting the `.terraform` folder has the downside that you need to run `terraform init` again next time.

Can you try this again and see if this is really necessary?

Comment (trcywu) [reply infrastructure/README.md:134]: Matt and I both got the same error when we were trying to spin up our clusters. When we went through and debugged it was because there was a cached version in .terraform so deleting it resolved the error. Can delete for now and if the error comes again at least we know what it is.

Comment (frigus02) [reply infrastructure/README.md:134]: Was your cached version still from our mobbing session? We changed the tfstate storage account after that. So that would be an explanation.

Comment (trcywu) [reply infrastructure/README.md:134]: ah yes, it was from the mobbing session.

Comment (frigus02) [reply infrastructure/README.md:134]: I'm pretty sure that was the reason. As long as we don't change the tfstate config anymore (which we won't 🤞), this should not be necessary.

Or: This step is only necessary if the backend config changed.

Let's remove this as an explicit step here. It could be a good idea to add a `## Troubleshooting` section at the end of this file, though. We can add this tip there.

File: infrastructure/environments/production/main.tf [hunk 5]
```diff
@@ -23,3 +23,10 @@ module platform {
   environment = "production"
   location    = var.location
 }
+
+resource "azurerm_container_registry" "acr" {
+  name                = "fnhsproduction"
+  resource_group_name = "platform-production"
```
Comment (frigus02) [inline infrastructure/environments/production/main.tf:29]: I think we need to add a dependency here. The resource group is created inside the platform module. But terraform is not smart enough to figure this out on it's own.

I suggest we add an `output` variable in the platform module, e.g.

```ts
output "resource_group_name" {
  value = azurerm_resource_group.platform.name
}
```

Then we can use it here:

```ts
   resource_group_name = module.platform.resource_group_name
```

File: infrastructure/README.md [hunk 6]
```diff
@@ -6,127 +6,131 @@ We use [Terraform](https://www.terraform.io/) to build our environments.
...
 1. Install Terraform:
 
-    ```bash
-    brew install terraform
-    ```  
-   
+   ```bash
+   brew install terraform
+   ```
+
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply infrastructure/README.md:12]: Hmm @trcywu @alsuren , sorry i'm late too this, i did not recieve an email notification, and have so only seen this in the notifications dashboard i'm starting to play with.  Firstly, is this still unresolved, and secondly i find the question confusing ( unsure relevence of IDE plugin ?) - could you rephrase / elaborate please if you still need input?
